Source/WebCore/page/Chrome.cpp

```diff
@@ -203,9 +205,9 @@ void Chrome::takeFocus(FocusDirection direction)
     m_client->takeFocus(direction);
 }
 
-void Chrome::focusedElementChanged(Element* element)
+void Chrome::focusedElementChanged(Element* element, Frame* frame, FocusOptions options, BroadcastFocusedElement broadcast)
 {
-    m_client->focusedElementChanged(element);
+    m_client->focusedElementChanged(element, frame, options, broadcast);
```

Comment:
IIUC, `frame` should either be equal to the `element`'s frame if it's a local frame; otherwise, if it's a remote frame, the `element` must be null? (If that's accurate, it might be good to add an assertion here to enforce that invariant).

Reply:
That's correct, I added assertions in a couple of places! (`Chrome:: focusedElementChanged() and `FocusController::setFocusedElement()`)